velox/type/TimestampConversion.cpp

```diff
@@ -812,6 +814,83 @@ Status parserError(const char* str, size_t len) {
       std::string(str, len));
 }
 
+inline bool startsWith(std::string_view str, const char* prefix) {
+  return str.rfind(prefix, 0) == 0;
+}
+
+bool parseSparkTzWithPrefix(
+    std::string& timeZoneName,
+    Timestamp& ts,
+    size_t start) {
+  VELOX_DCHECK_GT(timeZoneName.length(), start + 2);
+  auto sign = timeZoneName[start];
+  auto offset = timeZoneName.substr(start + 1);
+  std::string hm, h, m, s;
+  RE2 hhPattern(R"((\d?\d))");
```

Comment:
how hard would it be to implement the parsing logic without using re2, like we do for other timestamp conversions/parsing? I'm slightly worried about performance, but also adding re2 as a dependency to basically everything else in the library.

Reply:
The primary purpose here is to extract these four scenarios.
- h[h]
- hh[:]mm
- hh:mm:ss
- hhmmss

Using re2 code might make it more concise, but for performance considerations, I will try to implement the logic here without using re2.